Updating Developer Guide README.md to include Hybrid search specific documentation

adityabharadwaj198 opened the pull request:
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
* docs update


* **What is the current behavior?** (You can also link to an open issue here)
* Currently the developer guide readme doesn't contain steps to create the Vespa custom searcher's .jar file, which would lead to a contributor encountering issues when following the developer guide


* **What is the new behavior (if this is a feature change)?**
* Updated readme file contains the steps to create the .jar file.  


* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
* No


* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
* Unit tests aren't required for the PR 


* **Related Python client changes** (link commit/PR here)
* None


* **Related documentation changes** (link commit/PR here)
* None


* **Other information**:


* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

adityabharadwaj198 committed:
Updating Developer Guide README.md to include Hybrid search specific documentation

adityabharadwaj198 committed:
Update README.md

Fixed numbering of the steps

adityabharadwaj198 committed:
Update README.md

Fixed numbering again

farshidz commented on src/marqo/README.md:
```diff
@@ -35,14 +35,56 @@ docker run --detach --name vespa --hostname vespa-tutorial \
-
 You can verify that Vespa has been set up correctly by visiting `http://localhost:8080` in your browser.
 
+5. We've written custom logic to handle to reranking of all the results that Vespa outputs. We call this Hybrid search. Hybrid Search combines both lexical and tensor search to provide the best relevance possible. Hybrid search provides flexibility to get the best results of both types of search and fine tune which type to lean into. It's a great place to start when you want the precision of lexi ...
```
No need to refer to hybrid search here

adityabharadwaj198 replied:
removed hybrid search usage

farshidz commented on src/marqo/README.md:
````diff
@@ -35,14 +35,56 @@ docker run --detach --name vespa --hostname vespa-tutorial \
+5a. Install Java Development Kit: 
+You can install the Java Development Kit (JDK) by following the instructions [here](https://openjdk.org/install/). 
+You will need to set Java to Path, you can do this by 
+```zsh
````
Use bash to be consistent

adityabharadwaj198 replied:
changed to bash everywhere

farshidz commented on src/marqo/README.md:
````diff
@@ -35,14 +35,56 @@ docker run --detach --name vespa --hostname vespa-tutorial \
+JAVA_HOME="/Library/Java/JavaVirtualMachines/jdk-22.0.2.jdk/Contents/Home"
+PATH="${JAVA_HOME}/bin:${PATH}"
+```
+Remember to replace the "/Library/Java/JavaVirtualMachines/jdk-22.0.2.jdk/Contents/Home" with the path to your JDK.
````
You could provide links but no need to cover how to install java etc in detail here

adityabharadwaj198 replied:
added links to install java/maven & removed details on installing

adityabharadwaj198 committed:
Addressing comments on PR #955

This commit addresses comments made by @farshidz on PR: https://github.com/marqo-ai/marqo/pull/955/commits/544d33ea39103f006e2b3a0972a19293c0af2bc1

farshidz committed:
Merge branch 'mainline' into update-readme

farshidz commented on src/marqo/README.md:
````diff
@@ -35,14 +35,35 @@ docker run --detach --name vespa --hostname vespa-tutorial \
+```bash
+mvn -version
+```
+Post this you need to create a jar file, cd into the vespa directory in your local marqo repository, and run 
````
Make this step 6

adityabharadwaj198 replied:
done

farshidz commented on src/marqo/README.md:
````diff
@@ -35,14 +35,35 @@ docker run --detach --name vespa --hostname vespa-tutorial \
+mvn -version
+```
+Post this you need to create a jar file, cd into the vespa directory in your local marqo repository, and run 
+```bash
````
`cd vespa`?

adityabharadwaj198 replied:
added cd vespa in the line above

farshidz commented on src/marqo/README.md:
````diff
@@ -35,14 +35,35 @@ docker run --detach --name vespa --hostname vespa-tutorial \
+```bash
+mvn -version
+```
+Post this you need to create a jar file, cd into the vespa directory in your local marqo repository, and run 
````
The wording is confusing. How do I create a jar file?

adityabharadwaj198 replied:
changed

farshidz commented on src/marqo/README.md:
```diff
@@ -35,14 +35,35 @@ docker run --detach --name vespa --hostname vespa-tutorial \
 
-5. Open the Marqo project in your IDE (e.g. PyCharm) and go to the file `src/marqo/tensor_search/api.py`
-6. Set up your [debug configuration](https://www.jetbrains.com/help/pycharm/creating-run-debug-configuration-for-tests.html)
+6. Open the Marqo project in your IDE (e.g. PyCharm) and go to the file `src/marqo/tensor_search/api.py`
```
(e.g. PyCharm) once is enough. After that it's redundant

farshidz commented on src/marqo/README.md:
```diff
@@ -35,14 +35,35 @@ docker run --detach --name vespa --hostname vespa-tutorial \
+Post running this command, you will see that a target folder gets created in the vespa directory, which contains a jar file called marqo-custom-searchers-deploy.jar. This jar file is used to deploy the custom searchers to Vespa.
+
 ### Option A. Run the Marqo application locally (outside of docker) through IDE
 Now you can run Marqo locally through your IDE (e.g. PyCharm) by following the steps below.
```
e.g., (missing comma)

farshidz commented on src/marqo/README.md:
````diff
@@ -52,14 +73,14 @@ VESPA_CONFIG_URL=http://localhost:19071
 ```
-7. Now you can Debug this file directly from your IDE (e.g. PyCharm) to start Marqo locally.
-8. Set breakpoints in the project for better debugging experience.
+8. Now you can Debug this file directly from your IDE (e.g. PyCharm) to start Marqo locally.
````
debug (lower case)

adityabharadwaj198 committed:
[Addressing comments]

Addressing comments on PR commit: https://github.com/marqo-ai/marqo/pull/955/commits/ef985fd90e16d2859882e8f902f0c546de3d0840

farshidz committed:
Merge branch 'mainline' into update-readme

farshidz committed:
Merge branch 'mainline' into update-readme
